### ISSUE TITLE
[HOLD for payment 2023-05-03] [$1000] Screen is shaking when scrolling while user edit a comment between 2 tabs.

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#16931]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open a new report, try to send few message (Make sure the content of the message has over 15 lines), open that report in another tab.
2. From the first tab, try to edit a multiple lines message, make sure the content is scrollable.
3. From the second tab, delete all the content of that message, return the first tab try to scroll to bottom few times. 
4. Notice that screen is shaking while scrolling.

## Expected Result:
User should scroll to bottom without any issue.

## Actual Result:
Screen is shaking while scrolling.

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop
 
**Version Number:** 1.2.94-0
**Reproducible in staging?:** needs reproduction
**Reproducible in production?:** Needs reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/43996225/229883438-e444e35a-78ab-4b80-882e-2b4a8f8777f2.mov

**Expensify/Expensify Issue URL:**
**Issue reported by:** @hungvu193 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1680608255498089

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01c4cdb2f5c5e79041</li>
        <li>Upwork Job ID: 1645761324986847232</li>
        <li>Last Price Increase: 2023-04-18</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @laurenreidexpensify (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by laurenreidexpensify** [comment]
I''m finding this one pretty hard to reproduce based on the instructions above - @kavimuru @hungvu193 can you break this down a bit further? Thanks

**Comment by hungvu193** [comment]
@laurenreidexpensify I think I can give you more information here.
- Try to create a new conversation
- Make sure the content of the edit message is scrollable.


https://user-images.githubusercontent.com/16502320/230780845-7f0f6746-fac8-4555-9e17-7b3e24bb459f.mov

**Comment by MelvinBot** [comment]
@laurenreidexpensify Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01c4cdb2f5c5e79041

**Comment by MelvinBot** [comment]
Current assignee @laurenreidexpensify is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @aldo-expensify (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by aldo-expensify** [comment]
I'm having problems reproducing too

**Comment by hungvu193** [comment]
@aldo-expensify 
you can try this one:
```
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
https://staging.new.expensify.com/r/6006786908332036
https://staging.new.expensify.com/r/6006786908332036
https://staging.new.expensify.com/r/6006786908332036
https://staging.new.expensify.com/r/6006786908332036
https://staging.new.expensify.com/r/6006786908332036
```
And the evidence for latest staging:

https://user-images.githubusercontent.com/16502320/231634252-24eec7b6-649e-4c5b-9c84-098b00c37d1f.mov

**Comment by hungvu193** [comment]
@aldo-expensify I can reproduce everytime, I'm using mac touchpad.
Can you try these steps and let me know if it works?
1. Go to new report.
2. Chat 3,4 messages.
3. Send a very long message, just to make sure content of edit message is scrollable, like this:
```
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
[https://staging.new.expensify.com/r/6006786908332036](https://staging.new.expensify.com/r/6006786908332036)
```
4. Edit the message, delete all the content but don't save, now open that report in another tab.
5. Scroll inside your edit message then scroll outside.

https://user-images.githubusercontent.com/16502320/231931348-904135b6-9a3f-42f1-87e5-4c13851cb83b.mov

**Comment by aldo-expensify** [comment]
@hungvu193 ahh thanks, now I reproduced it. I had completely missed the stuff about using 2 tabs 🤦

**Comment by laurenreidexpensify** [comment]
@hungvu193 any proposal for how to fix is 😃?

**Comment by hungvu193** [comment]
@laurenreidexpensify I think we can sync the edit message between 2 tabs. This issue won't happen if the edit message between 2 tabs is synchronize.
We actually synced the draft message between 2 tabs.

**Comment by s77rt** [comment]
@hungvu193 Seems like you have a valid proposal. Can you please follow the proposal template?

**Comment by MelvinBot** [comment]
@s77rt @laurenreidexpensify @aldo-expensify this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by hungvu193** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Screen is shaking when scrolling while user edit a comment between 2 tabs.
### What is the root cause of that problem?
We are sharing `numberOfLines` through Onyx, but not the draft message.
When we tried to change the draft between the tab, the `numberOfLines` between two tabs now are different, it will cause an infinite loop between two tabs to change the numberOfLines. This thing will cause the shaking that we saw in the issue.

https://user-images.githubusercontent.com/16502320/232745653-f7f07570-7dfa-4b74-bc22-e1356c485032.mov


### What changes do you think we should make in order to solve the problem?
We have 2 options here:
- Make the draft sync between 2 tabs (similar to the comment). (We also need to handle the case that user delete all the draft but didn't press Cancel button to close the Edit session).
- Since we didn't sync the edit draft message between 2 tabs, we also shouldn't share the numberOfLines between them as well. So we can remove the `updateNumberOfLines` props from `ReportActionItemMessageEdit`.

### What alternative solutions did you explore? (Optional)

**Comment by MelvinBot** [comment]
📣 It's been a week! Do we have any satisfactory proposals yet? Do we need to adjust the bounty for this issue? 💸

**Comment by s77rt** [comment]
@hungvu193 Thanks for the proposal. I think you are right about the diff on `numberOfLines` causing the bug. I have a non-blocking question, why exactly the scroll bug occurs? Is it because we are rendering the input many times, or is this related to virtual list somehow? etc,..

And I have a blocking question :sweat_smile: Can you please elaborate how you are going to sync the edit comment and how you are going to fix the concern raised here https://github.com/Expensify/App/pull/15610#issuecomment-1452199924?

**Comment by hungvu193** [comment]
> @hungvu193 Thanks for the proposal. I think you are right about the diff on `numberOfLines` causing the bug. I have a non-blocking question, why exactly the scroll bug occurs? Is it because we are rendering the input many times, or is this related to virtual list somehow? etc,..
I think it related to focus and re-rendering, while you tried to scroll but your input is still focus and it's rerendering, it will focus on your input again, that's why we saw the shaking.



> And I have a blocking question 😅 Can you please elaborate how you are going to sync the edit comment and how you are going to fix the concern raised here [#15610 (comment)](https://github.com/Expensify/App/pull/15610#issuecomment-1452199924)?

Yeah that's right, we need to add another variable to handle the case when user try to delete all the draft, but they didn't press Cancel button yet, we also need to test really carefully. IMO, I prefer the second options, if we don't sync the draft value, we shouldn't sync the numberOfLines.

**Comment by s77rt** [comment]
@hungvu193 I'm not opposed to removing `numberOfLines` but we should investigate why it was added in the first place by https://github.com/Expensify/App/pull/15296. I think it's useless since `initWithStoredValues` is `false`.

cc @thesahindia @Pujan92 Some help please ^

___

@thesahindia I see you mentioned the bug here https://github.com/Expensify/App/pull/15296#issuecomment-1458630740 that bug still exists.
@Pujan92 Did we really had to modify the number of lines for the edit comments too?


https://user-images.githubusercontent.com/16493223/233034999-8237a9c0-f0b2-40ae-bc71-d260cfbd9ff0.mp4

**Comment by Pujan92** [comment]
> @Pujan92 Did we really had to modify the number of lines for the edit comments too?

@s77rt I think we do required bcoz of the same distortion issue, still I will do check more after some time once I reached to home.

**Comment by Pujan92** [comment]
> @Pujan92 Did we really had to modify the number of lines for the edit comments too?

@s77rt as for edit comment `initWithStoredValues: false` isn't helped for the first render but modification of number of lines do required for this case https://github.com/Expensify/App/pull/15296#issuecomment-1437508868.

**Comment by s77rt** [comment]
@Pujan92 The case you linked does not look to be related to the edit comment. Am I missing something here?
@thesahindia I understand that, but then there is no use storing the numberOfLines on onyx.

**Comment by Pujan92** [comment]
That Video shows scenario for the new comment composer but same is applied to edit comment too, without numberOfLines for this case edit comment composer will show default lines instead of the updated ones.

**Comment by s77rt** [comment]
@Pujan92 Thank you! That was helpful. I think that bug will no longer be reproducible after navigation reboot. Check below video after applying changes from https://github.com/Expensify/App/pull/16581 and removing `numberOfLines`:

https://user-images.githubusercontent.com/16493223/233239602-f38f0e34-4323-4d2d-ae35-45653ecd909f.mp4


___

Was that the sole purpose for adding `numberOfLines`? Do you think we can remove it now?

cc @thesahindia

**Comment by MelvinBot** [comment]
Looks like something related to `react-navigation` may have been mentioned in this issue discussion.

As a reminder, please make sure that all proposals are not workarounds and that any and all attempt to fix the issue holistically have been made before proceeding with a solution. Proposals to change our `DeprecatedCustomActions.js` files should not be accepted.

Feel free to drop a note in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q) with any questions.

**Comment by hungvu193** [comment]
@s77rt this PR (https://github.com/Expensify/App/pull/16581) is merged, I think we can remove numberOfLines.

**Comment by s77rt** [comment]
@hungvu193 Can you please test and confirm the bug reported here https://github.com/Expensify/App/issues/16931#issuecomment-1515235671 will no longer be reproducible?

@Pujan92 @thesahindia a confirmation please on https://github.com/Expensify/App/issues/16931#issuecomment-1515614966

**Comment by Pujan92** [comment]
> @Pujan92 @thesahindia a confirmation please on [#16931 (comment)](https://github.com/Expensify/App/issues/16931#issuecomment-1515614966)

Yes, as it is not producible after navigation reboot then we can remove it.

**Comment by s77rt** [comment]
@hungvu193 Thanks once again for your proposal and for the follow up :+1:. Let's go with your suggested fix option 2 (remove `numberOfLines` and `onNumberOfLinesChange` from `ReportActionItemMessageEdit`).

:ribbon: :eyes: :ribbon: C+ reviewed

cc @aldo-expensify

**Comment by aldo-expensify** [comment]
Thanks for the thorough investigation! I agree with removing this stuff if it is no longer needed and adds unnecessary complexity + bugs.

**Comment by hungvu193** [comment]
Thanks guys, PR is ready @s77rt @aldo-expensify

**Comment by laurenreidexpensify** [comment]
Offers sent in Upwork for @hungvu193 and @s77rt

**Comment by hungvu193** [comment]
@laurenreidexpensify Thank you. I've just applied

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.3.5-6 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/17899


If no regressions arise, payment will be issued on 2023-05-03. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@s77rt] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@s77rt] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@s77rt] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@s77rt] Determine if we should create a regression test for this bug.
- [ ] [@s77rt] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@laurenreidexpensify] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by s77rt** [comment]
- The PR that introduced the bug has been identified: https://github.com/Expensify/App/pull/15296
- The offending PR has been commented on: https://github.com/Expensify/App/pull/15296#issuecomment-1523171633
- A discussion in #expensify-bugs has been started: https://expensify.slack.com/archives/C049HHMV9SM/p1682504657797639
- Determine if we should create a regression test for this bug: Yes

___

**Regression Test Proposal** (Web Only)
1. Login to NewDot in 2 tabs 
2. Open same chat in both tabs (chat must be scrollable i.e. have previous messages)
5. **Tab 1**: Send a long multiline message e.g. 25 lines
6. **Tab 1**: Edit the message: delete the content but do not press the 'Save' button
7. **Tab 2**: Verify the message is in edit mode
8. **Tab 2**: Verify the message content is not cleared
9. **Tab 2**: Scroll inside the edit message 
10. **Tab 2**: Scroll outside the edit message
11. **Tab 2**: Verify that you were able to scroll smoothly

**Comment by laurenreidexpensify** [comment]
great work everyone!
